pkg/controller/operators/catalog/operator.go

```diff
@@ -813,12 +818,12 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 			}
 		}
 
-		installPlanReference, err := o.ensureInstallPlan(logger, namespace, subs, installPlanApproval, steps, bundleLookups)
+		installPlanReference, err := o.ensureInstallPlan(logger, namespace, maxGeneration+1, subs, installPlanApproval, steps, bundleLookups)
```

Comment:
can this use `updatedSubs` instead of `subs`?

Reply:
It's probably best to use `subs` since we want to apply `OwnerReferences` to the generated `InstallPlan` referencing all `Subscriptions` in the namespace. I think we also want to use `subs` a few lines below for the `Subscription` update so they point to the latest `InstallPlan`.

I'm probably missing some edge case though; e.g. bad `Subscriptions` should reference the latest `InstallPlan`.